### ISSUE TITLE
NVDA turns focus mode on and off when navigating between text inputs hosted in different iframes in Chrome

### ISSUE THREAD

**Issue by ochalet-wp** [nvaccess/nvda#12722]
Let's take the example of a page with 3 fields hosted in 3 different iframes:
- When navigating to the first text input (using tab or by clicking into the field) I can type in text (NVDA has switched on focus mode automatically)
- When navigating to the 2nd text input, I cannot type in text (NVDA has switched focus mode off)
- When navigating to the 3rd text input, I can type in text (NVDA has switched focus mode back on)

### Steps to reproduce:
1. Decompress attached zip file
2. In a terminal, `cd` into the directory where files have been decompressed
3. Run `npm init` and follow the invite to complete initialisation
4. Run `npm install express`
5. Run `node server.js`
6. Start NVDA and navigate to [http://localhost:3000/](http://localhost:3000/) in Chrome
7. Wait for NVDA to have fully read the content of the page
8. Click into Field 1
9. Attempt to enter text and confirm that you can enter text
10. Click into Field 2
11. Attempt to enter text and confirm that you cannot enter text
12. Click into Field 3
13. Attempt to enter text and confirm that you can enter text

### Actual behavior:
NVDA turns off form mode for Field 2 and I cannot enter text
[archive.zip](https://github.com/nvaccess/nvda/files/6956243/archive.zip)

### Expected behavior:
NVDA leaves form mode on for Field 2 and I can enter text.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
Release 2021.1

#### Windows version:
Windows 10 Version 2004 (OS Build 19041.1110)

#### Name and version of other software in use when reproducing the issue:
Chrome version 92.0.4515.131

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

#### If add-ons are disabled, is your problem still occurring?
Yes

**Comment by feerrenrut** [comment]
Hi @ochalet-wp, thanks for the report, however please provide a minimal example. Having to install and run node to demonstrate something like this is excessive. From the description it should be reproducible with a simple HTML example. You could consider making a codePen or jsBin for the example.

**Comment by ochalet-wp** [comment]
Hi @feerrenrut, thank you for the feedback. I have updated the steps to reproduce the issue with a simple HTML example.

**Comment by feerrenrut** [comment]
Thanks @ochalet-wp, I have tested this with Edge (Chromium) and NVDA 2020.1. Using tab to move between fields I get the following speech:
```
Field 1  edit  Enter value 1  blank
Focus mode
frame    clickable  Field 2  edit    Enter value 2
Browse mode
Field 3  edit  Enter value 3  blank
Focus mode
```

The changes between focus and browse mode seem to alternate depending on what mode NVDA is in first, this doesn't seem to be anything specific to the controls.

Note: @ochalet-wp, when NVDA is in browse mode, it captures keyboard input, the keys become shortcuts to make navigation easier. For-instance `h` jumps to the next heading. By default browse mode is signified via an earcon "boop" sound, focus mode has a "click" sound. In browse mode settings these sounds can configured to use speech. A user can manually toggle between the modes with `NVDA+space`.

**Comment by ochalet-wp** [comment]
Hi @feerrenrut, thank you for the detailed analysis. 

I understand that NVDA automatically switches on focus mode when the user navigates to an editable field (although that option can be switched off by the user). So isn't it a case that in our example focus mode should remain turned on when moving the focus from Field 1 to Field 2?

**Comment by feerrenrut** [comment]
Yes, this sounds like a problem. I suspect the difficulty comes from the iframes.

In case the original sample disappears, here is a codepen: https://codepen.io/reefturner/pen/MWoGgda

**Comment by Adriani90** [comment]
I tag here @ObjectInSpace and @aleventhal since this issue seems to be chrome specific.

**Comment by ObjectInSpace** [comment]
I haven't worked at Chrome for some time. (If you have any WhatsApp issues, I'm your guy!) I think Aaron does not check github much. If it's an issue in Chrome your best bet is to file it on crbug.com.
